### DISCUSSION TITLE
Unable to serve `.css` and `.js` – FileServer required?

### DISCUSSION THREAD

**Discussion by Glitchy-Tozier** [rwf2/Rocket#2159] [Questions] [answered]
I'm working on a web-server that uses dynamic paths (`#[get("/<poll_id>")]`). Now I've got the problem that my `static/index.html` (exposed via `#[get("/")]`) doesn't seem to have access to `static/main.css`.
(If I remove the `#[get("/")]`, i can't find `index.html` anywhere. `/`, `/index.html`, and `/static/index.html` don't work.)

Am I still supposed to use this command `.mount("/", FileServer::from("static/"))` described [here](https://rocket.rs/v0.5-rc/guide/requests/#multiple-segments), even though my server is more than a static file-server?

What is the intended solution for this (probably _extremely common_) usecase?

**Comment by SergioBenitez** [comment, marked as answer]
Yes. These are completely orthogonal things. It's not one or the other, it's both, if that's what you need. At the end of the day, they're all just routes.

**Reply by Glitchy-Tozier** [reply]
I'll try that out, then, thank you!
Is there any particular behaviour I should be aware of regarding [ranking](https://rocket.rs/v0.5-rc/guide/requests/#default-ranking) of static vs non-static?

**Reply by SergioBenitez** [reply]
> I'll try that out, then, thank you!
> 
> Is there any particular behaviour I should be aware of regarding [ranking](https://rocket.rs/v0.5-rc/guide/requests/#default-ranking) of static vs non-static?

These really are all just routes. There's nothing special about the routes mounted by the FileServer struct. They do have a ranking, which is customizable and defaults to 10.
